### DISCUSSION TITLE
Please! How to slove this problem?

### DISCUSSION THREAD

**Discussion by TheMyNameABC** [shap/shap#3480] [Q&A]
my SHAP version is 0.43.0

I found a warning when I run this code:
 start_time = time.time()
 explainer = shap.SamplingExplainer(stacking_model.predict, X_train, link="identity")
 shap_values = explainer.shap_values(X_test, n_jobs=14) 
 end_time = time.time()

 elapsed_time_minutes = (end_time - start_time) / 60
 print(f"Total elapsed time: {elapsed_time_minutes:.2f} minutes")

warning: Usage of np.ndarray subset (sliced data) is not recommended due to it will double the peak memory cost in LightGBM.

my Stacking model is combined by LightGBM+XGBoost+SVR+ENet
the code runs very slowly
Please tell me how to slove it.
Thanks
<img width="722" alt="微信图片_20240204235724" src="https://github.com/shap/shap/assets/115081759/884849d4-56ff-4122-9e4c-f20d07f1134d">

**Comment by CloseChoice** [comment]
This is an xgboost warning. See https://stackoverflow.com/questions/67225016/warning-occuring-in-xgboost or https://github.com/dmlc/xgboost/issues/6908#issuecomment-834731526 for reference.
